holoviews/core/data/dictionary.py

```diff
@@ -56,7 +56,8 @@ def init(cls, eltype, data, kdims, vdims):
             data = {k: data[:,i] for i,k in enumerate(dimensions)}
         elif isinstance(data, list) and np.isscalar(data[0]):
             data = {dimensions[0]: np.arange(len(data)), dimensions[1]: data}
-        elif not isinstance(data, dict):
+        elif not any(isinstance(data, tuple(t for t in interface.types if t is not None))
```

Comment:
I would just add a comment to say this is needed as dictionary can incorrectly accept xarray (duck typing fails here!). The comment might also mention that a cleaner approach would be welcome...